docs/release_new_version.md

```diff
@@ -0,0 +1,96 @@
+with `tmux at` (`Ctrl+B, D` by default to later dettach instead of closing the connection).
+
+Now run `ce --env staging builds list` and make sure that your target `gh-xxxx` build is present.
+
+_If this is not the case, make sure that there were no errors in the CI, or ask around for help._
+
+## Run compiler discovery of your target build
+
+Run a "Compiler Discovery" workflow in https://github.com/compiler-explorer/infra/actions/workflows/compiler-discovery.yml
+
+You need to fill 3 inputs:
+ - **Environment**: This is usually `staging`, but can be run for `beta` and `prod` too
+ - **Branch**: The branch that your target commit belongs to. Usually `main`
+ - **Build number**: The `gh-xxxx` id you identified 
+ 
+It usually takes less than 5 mins for the discovery to run.
+
+## Set current version in staging
+
+Once the discovery has finished, you can proceed to set the build as the current release version in staging.
+(The idea is to first test in staging as a last check before bouncing the version in prod)
+
+Once there, `ce --env staging builds set_current gh-xxxx` will set the current build version
+for staging to your identified build id.
+
+Note that `--env staging` comes before any command, and its default value is always `staging`,
+but uses it explicitly to be more clear as to what is running for which environment.
+
+After passing the sanity checks (it will ask you to confirm what branch this comes from and in what env you're running the command in), the build has been marked as current.
+Note that these sanity checks are present in most commands,
+and we're always on the lookout for more places to put them, so if you find anything not secured by these checks,
+please do let us know.
+
+_If this fails due to some hash missmatch errors, you need to bump the hack version number in `webpack.config.esm.js`._
+
+## Bringing staging up
+
+Now staging needs to be brought up. This is done with `ce --env staging environment start`.
+The message should be that the number of instances has been increased from 0 to 1 (Or more!)
+
+If this is not the case, it means that staging was already up and you'll need to instead refresh those instances.
+This is acomplished by running `ce --env staging environment restart`.
+This can take a bit until all the new nodes are healthy.
+
+Once this is done, the new version is running in https://godbolt.org/staging
+(You can confirm it by going to Other > Version Tree, which points to the release commit in GitHub)
+
+## Bringing staging down
+
+Unless staging is currently in use for something else, you should now always remember to bring staging down.
+
+This is unsuprisingly acomplished by running `ce --env staging environment stop`.
+
+## Mark discovery run as safe for production
+
+Instead of now rerunning the first step of compiler discovery for your commit but now for the prod environment,
+you can instead run `ce --env staging safeforprod gh-xxxx`, and it will create a compiler discovery result for prod.
```

Comment:
Not sure of the syntax for this one, will check tomorrow

Reply:
`ce runner safeforprod gh-xxxx`

Reply:
Looks like it might be `ce runner safeforprod staging gh-xxxx`

Reply:
^- still needs to be tweaked (you mention `ce runner` below but not here)
```suggestion
you can instead run `ce --env staging runner safeforprod gh-xxxx`, and it will create a compiler discovery result for prod.
```